From 0685f51a859bdbee572da7e2be53c8c74528106c Mon Sep 17 00:00:00 2001
From: Jun Yang <jun.yang@suse.com>
Date: Fri, 3 Dec 2021 12:01:50 +0800
Subject: [PATCH] fix hns_roce_main.c compilation error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch fixes below compilation error:

  CC [M]  drivers/infiniband/hw/hns/hns_roce_main.o
drivers/infiniband/hw/hns/hns_roce_main.c:431:3: error: ‘const struct ib_device_ops’ has no member named ‘create_user_ah’; did you mean ‘create_ah’?
  .create_user_ah = hns_roce_create_ah,
   ^~~~~~~~~~~~~~
   create_ah
drivers/infiniband/hw/hns/hns_roce_main.c:431:20: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
  .create_user_ah = hns_roce_create_ah,
                    ^~~~~~~~~~~~~~~~~~

mainline patch 66d86e529dd58616495ea0b03cc687e5d6522b59
v5.11-rc1-RDMA-hns-Add-UD-support-for-HIP09.patch
add create_user_ah for hns, but openeuler didn't use it.
It's hard to port all mainline patch to add create_user_ah,
so remove it in this patch.

Signed-off-by: Jun Yang <jun.yang@suse.com>
---
 drivers/infiniband/hw/hns/hns_roce_main.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/infiniband/hw/hns/hns_roce_main.c b/drivers/infiniband/hw/hns/hns_roce_main.c
index ac67b3e..0cd7ad5 100644
--- a/drivers/infiniband/hw/hns/hns_roce_main.c
+++ b/drivers/infiniband/hw/hns/hns_roce_main.c
@@ -428,7 +428,6 @@ static const struct ib_device_ops hns_roce_dev_ops = {
 	.alloc_pd = hns_roce_alloc_pd,
 	.alloc_ucontext = hns_roce_alloc_ucontext,
 	.create_ah = hns_roce_create_ah,
-	.create_user_ah = hns_roce_create_ah,
 	.create_cq = hns_roce_create_cq,
 	.create_qp = hns_roce_create_qp,
 	.dealloc_pd = hns_roce_dealloc_pd,
-- 
2.31.1

